src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -577,6 +577,18 @@ private static bool CanReplace(ISymbol symbol)
             return false;
         }
 
+        public static bool IsTopmostExpression(this ExpressionSyntax node)
+        {
+            return node.WalkUpParentheses().Parent switch
+            {
+                LambdaExpressionSyntax _ => true,
+                AssignmentExpressionSyntax _ => true,
+                ConditionalExpressionSyntax _ => true,
+                ExpressionSyntax _ => false,
+                _ => true
```

Comment:
doc this.  i don't really get it.  i'm not even sure what "TopmostExpression" means.  this feels like this is a helper for a particular feature, not a general helper.